### PR TITLE
 fix #15188 replaced isnan with std::isnan to avoid build error 

### LINKED ISSUE #15188: error: 'isnan' was not declared in this scope
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: Master branch commit: `4ad12049`
- **Python version**: N/A
- **Bazel version (if compiling from source)**: 
- **GCC/Compiler version (if compiling from source)**: g++ 5.4.0
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: `bazel build //tensorflow:libtensorflow.so`

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

I'm trying to compile using `bazel build //tensorflow:libtensorflow.so`.
I'm getting this error:
```
ERROR: /home/rdi/Workspace/Common/tensorflow/tensorflow/core/BUILD:577:1: C++ compilation of rule '//tensorflow/core:random_ops_op_lib' failed (Exit 1)
In file included from ./tensorflow/core/framework/allocator.h:23:0,
                 from ./tensorflow/core/framework/tensor.h:20,
                 from ./tensorflow/core/framework/attr_value_util.h:24,
                 from ./tensorflow/core/framework/node_def_util.h:23,
                 from ./tensorflow/core/framework/shape_inference.h:20,
                 from ./tensorflow/core/framework/common_shape_fns.h:20,
                 from tensorflow/core/ops/random_ops.cc:16:
./tensorflow/core/framework/numeric_types.h: In constructor 'tensorflow::bfloat16::bfloat16(float)':
./tensorflow/core/framework/numeric_types.h:49:16: error: 'isnan' was not declared in this scope
     if (isnan(v)) {
                ^
```

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by lordadamson** [comment]
I signed it!

**Comment by yongtang** [comment]
See comment https://github.com/tensorflow/tensorflow/commit/d87a76d7d6b053219dbd49a87b3c4b379a1c6566#r26118028

Replacing `isnan` with `std::isnan` may have issues with nvcc, maybe you could try adding a
```
using namespace std;
```
 and see if it solved the issue?

/cc @hawkinsp @lamerman

**Comment by hawkinsp** [comment]
Using std::isnan causes a runtime crash on a CUDA build with nvcc — I've tried that before. I can't reproduce your failure (maybe it is compiler version dependent).

Can you try using Eigen::numext::half for me? Does that compile for you?

File: tensorflow/core/framework/numeric_types.h [hunk 1]
```diff
@@ -46,7 +46,7 @@ struct bfloat16 {
   EIGEN_DEVICE_FUNC bfloat16() {}
 
   EIGEN_DEVICE_FUNC explicit bfloat16(const float v) {
-    if (isnan(v)) {
+    if (std::isnan(v)) {
```
Comment (gunan) [inline tensorflow/core/framework/numeric_types.h]: Would it be better to use `Eigen::numext::isnan` here, would that be a better pick for on-device code?
@hawkinsp @zheng-xq

**Comment by gunan** [comment]
@hawkinsp, to reproduce the problem you will need to try to build with cuda9:
https://github.com/tensorflow/tensorflow/pull/14773

Since we know that `std:isnan` wont work, I will push the recommended update to this PR.

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by gunan** [comment]
Jenkins, test this please.

**Review by gunan — APPROVED** [review]

**Comment by lordadamson** [comment]
Thank you all <3

**Comment by hawkinsp** [comment]
Sorry for the breakage and thanks for the PR!

**Pull request merged**